solucao-de-problemas-service/src/main/java/com/ages/incuitech/backend/solucaodeproblemasservice/business/cliente/ClienteService.java

```diff
@@ -59,11 +66,28 @@ public Cliente salvar(Cliente cliente) {
                         mapping(UserTag::getTagName, toList())));
     }
 
+    private List<Tag> salvarTags(List<String> tags) {
+        return tags.stream()
+                .map(tagService::salvar)
```

Comment:
Tu está indo no banco pra cada cara da tua lista. O correto é fazer uma inserção em batch.
Links que podem ajudar:
- https://www.baeldung.com/spring-jdbc-jdbctemplate#batch-operations
- https://www.tutorialspoint.com/springjdbc/springjdbc_batch_operation.htm
- https://docs.spring.io/spring/docs/3.0.0.RC2/reference/html/ch12s04.html

Reply:
Entendo os moivos de fazer inserção em lote de uma lista no banco, de que vai poupar processamento do banco de dados alem de não precisar ficar fazendo muitas operações de I/O.

Mas fico pensando se é relamente necessario criar um novo método que faça todo a operação que o salvar já faz com os seus devidos ajustes para que seja possivel salvar como `saveAll`do repositories, e, consequentemente, mudar toda a forma que as tags são salvas, tanto no cliente service quanto do solucionador service.

Acho valido, mas não acho necessario fazer essa mudança na altura do campeonato

Reply:
Coloquei vários links explicando como fazer, e é bem simples e rápido, e feito em poucas linhas de código. Se fosse um refactor gigante, aí eu veria sentido em talvez deixar como débito técnico.
Outro ponto é que eu não imagino algo mais impactante que possa ser pego durante o Code Review do que isso. Se nem isso é necessário refatorar, então eu não vejo qual é a utilidade do CR. Qual é o objetivo de fazermos CR? É pra refatorar questões estilo, ou questões que realmente impactam o sistema, como performance, design, segurança, etc. Se for só pra pegar questões de estilo, aí sim eu vejo que o CR é uma perda de tempo. Se for pra melhorar o design do sistema, aí não vejo porquê deixar um refactor simples e impactante como esse passar.
Mas se o ponto é não ser mais necessário refactor, e não ser necessário CR, isso tem que ser levantado e definido com o time primeiro. Por enquanto nós ainda temos essa etapa no nosso processo que deve ser respeitada.

Reply:
Acho válida a correção, claramente melhoraria a performance, mas será que vale a pena a gente mudar isso tão drasticamente hj no dia da entrega? Se está funcionando assim hoje, que é o que a gnt precisa para entregar a US pro cliente, acho melhor manter do que arriscar não entregar a US por conta da melhoria de performance, (que pode sim ser feita, mas dps da entrega de hj). So deixando a minha opinião. Independente do que for decidido, hoje anoite a gnt pode falar melhor sobre isso também.

Reply:
Sabemos que isso tá funcionando? Foi validado no ambiente? Não tinha nenhum teste antes de eu pedir. Me parece arriscado de qualquer forma.